server/src/main/kotlin/org/jetbrains/bsp/bazel/server/sync/ClasspathQuery.kt

```diff
@@ -15,8 +16,14 @@ object ClasspathQuery {
                 .executeBazelCommand(parseProcessOutput = false)
                 .waitAndGetResult(cancelChecker, ensureAllOutputRead = true)
         if (cqueryResult.isNotSuccess) throw RuntimeException("Could not query target '${target.uri}' for runtime classpath")
-        val classpaths = Gson().fromJson(cqueryResult.stdout, JvmClasspath::class.java)
-        return classpaths
+        try {
+            val classpaths = Gson().fromJson(cqueryResult.stdout, JvmClasspath::class.java)
+            return classpaths
+        } catch (e: MalformedJsonException){
+            // sometimes Bazel returns two values to a query, which are not parseable, the real query should be oneline
+            return if (cqueryResult.stdoutLines.size > 1) Gson().fromJson(cqueryResult.stdoutLines[0], JvmClasspath::class.java)
```

Comment:
Can you show example output from this cquery command? 
I'd guess that there are simply 2 configurations. It can happen when for example docker rules or something similar is in the project.

Reply:
When I run it separately, I only get one result :/

Reply:
hmm, it is a bazel command, it should be reproducible :p if you print the command that bazel runner runs and put a breakpoint on the catch here, you should get exact command. It doesn't reproduce the issue?

Reply:
You might need to do a clean build and import of bazel-bsp and install server again to be sure that sources match the jars.

Reply:
it was my bad, wrong exception is being caught

Reply:
For example I am now working on multiversion scala and the lib and dep-lib target, despite being defined once, has 3 configurations, as it is a dependency of 3 targets with different scala version, bazel creates a separate configuration for each.
```sh
% bazel cquery //example/...
INFO: Analyzed 5 targets (0 packages loaded, 0 targets configured).
INFO: Found 5 targets...
//example:dep-lib (3b27016)
//example:dep-lib (6dfe423)
//example:dep-lib (b62978e)
//example:lib (3b27016)
//example:lib (6dfe423)
//example:lib (b62978e)
//example:myprog_on_scala_1 (6dfe423)
//example:myprog_on_scala_2 (b62978e)
//example:myprog_on_scala_3 (3b27016)
```

In case that @tgodzik is investigating, it is connected to packaging/dockerizing rules.
In fact, it is not fully deterministic:

query for 1 target and get 1 configuration (after bazel clean)
```
% bazel cquery  //persistence:persistence-lib
(13:34:47) INFO: Current date is 2024-02-01
(13:34:52) INFO: Analyzed target //persistence:persistence-lib (221 packages loaded, 4998 targets configured).
(13:34:52) INFO: Found 1 target...
//persistence:persistence-lib (aacd9ab)
```

query with wildcard and get 2 configurations (and more other targets that I skipped, some have 2 configs, some have 1) 
```sh
% bazel cquery  //persistence/...
//persistence:persistence-lib (aacd9ab)
//persistence:persistence-lib (HOST)
```

query the single target again and get 2 configurations (until another bazel clean):
```
% bazel cquery  //persistence:persistence-lib
(13:38:42) INFO: Current date is 2024-02-01
(13:38:42) INFO: Analyzed target //persistence:persistence-lib (0 packages loaded, 0 targets configured).
(13:38:42) INFO: Found 1 target...
//persistence:persistence-lib (aacd9ab)
//persistence:persistence-lib (HOST)
(13:38:42) INFO: Elapsed time: 0.128s
(13:38:42) INFO: 0 processes.
(13:38:42) INFO: Build completed successfully, 0 total actions
```

Reply:
Ideally, each configuration would be treated as a separate target, but that seems like it could be an overkill here. Otherwise we need some kind of heuristic here such as:
- take first configuration
- take the configuration with most classpath entries
- sort it via some means

What do you think @tpasternak ?